--- a/arch/arm/mach-mvebu/Kconfig
+++ b/arch/arm/mach-mvebu/Kconfig
@@ -128,4 +128,11 @@
 	  Say 'Y' here if you want your kernel to support boards based
 	  on the Marvell Kirkwood device tree.
 
+config KIRKWOOD_ALWAYS_RESTART
+	bool "Restart board when powering off kernel"
+	depends on MACH_KIRKWOOD
+	help
+	  Say 'Y' here if you want your kernel to restart
+	  when asked to power off.
+
 endif
--- a/arch/arm/kernel/reboot.c
+++ b/arch/arm/kernel/reboot.c
@@ -102,9 +102,13 @@
  */
 void machine_halt(void)
 {
+#ifdef CONFIG_KIRKWOOD_ALWAYS_RESTART
+	machine_restart(NULL);
+#else
 	local_irq_disable();
 	smp_send_stop();
 	while (1);
+#endif
 }
 
 /*
@@ -115,9 +119,13 @@
  */
 void machine_power_off(void)
 {
+#ifdef CONFIG_KIRKWOOD_ALWAYS_RESTART
+	machine_restart(NULL);
+#else
 	local_irq_disable();
 	smp_send_stop();
 	do_kernel_power_off();
+#endif
 }
 
 /*
